salt/modules/mac_shadow.py

```diff
@@ -0,0 +1,391 @@
+# -*- coding: utf-8 -*-
+'''
+.. versionadded:: 2016.3.0
+
+Manage Mac OSX local directory passwords and policies.
+
+Note that it is usually better to apply password policies through the creation
+of a configuration profile.
+'''
+# Authentication concepts reference:
+# https://developer.apple.com/library/mac/documentation/Networking/Conceptual/Open_Directory/openDirectoryConcepts/openDirectoryConcepts.html#//apple_ref/doc/uid/TP40000917-CH3-CIFCAIBB
+
+from __future__ import absolute_import
+from datetime import datetime
+
+# Import salt libs
+import salt.utils
+import logging
+
+log = logging.getLogger(__name__)  # Start logging
+
+__virtualname__ = 'shadow'
+
+
+def __virtual__():
+    # Is this os x?
+    if not salt.utils.is_darwin():
+        return False, 'Not Darwin'
+
+    return __virtualname__
+
+
+def _get_dscl_data_value(name, key):
+    '''
+    Return the value for a key in the user's plist file
+    For use by this module only
+
+    Args:
+        name: username
+        key: plist key
+
+    Returns:
+        the value contained within the key
+    '''
+    cmd = 'dscl . -read /Users/{0} {1}'.format(name, key)
+    ret = __salt__['cmd.run'](cmd)
+    if ': ' in ret:
+        value = ret.split(': ')[1]
+        return value
+    if ':\n' in ret:
+        value = ret.split(':\n')[1]
+        return value
+    return ret
+
+
+def _get_account_policy_data_value(name, key):
```

Comment:
There looks like a lot of code duplication between this function and the previous. Perhaps these could be refactored?